client/python/gradio_client/client.py

```diff
@@ -93,6 +109,67 @@ def fn(future):
 
         return job
 
+    def usage(self, all_endpoints=True, print_usage=True) -> str | None:
```

Comment:
I think setting `all_endpoints=True` as the default will generate a lot of noise for developers. The `Dataset.click` event will be shown for every interface, for example.

We should start encouring app authors to document the methods they intend to be used via api via api_name (or some other mechanism?)

Reply:
Yup but the reason is that most developers who are working with Blocks don't add api names. As a result, people don't even know the Space *could* be used as an API, which I think is a bigger problem.

One thing we could do is by default set `all_endpoints=False`, but then show at the bottom: "3 unnamed endpoints. To show usage for unnamed endpoints, set `all_endpoints=True`. WDYT?

Reply:
Or another idea is that if `all_endpoints` is not explicitly provided, we show all endpoints if there are no named endpoints.